### ISSUE TITLE
edit (FE)

### ISSUE THREAD

**Issue by MariusJulius** [keeleinstituut/tv-tolkevarav#242]

**Comment by KaarelKa** [comment]
Hey @thenouan @MariusJulius @PlaksoBirgit 

Had a question about cancelling an order

Basically input for it is combined from:

1.https://github.com/keeleinstituut/tv-tolkevarav/wiki/8.-Tellija-vaade-(Tellimuste-loomine,-korraldamine-ja-t%C3%B5lkimine-)#55-flow-cancel-project

2. https://github.com/keeleinstituut/tv-tolkevarav/wiki/A-11.-T%C3%B5lkekorraldaja-vaade-(Tellimuste-loomine,-korraldamine-ja-t%C3%B5lkimine-)#66-flow-cancel-project


Question:

In the wiki there is a sentence "Depending on the privileges, the user can cancel personal projects and also Institutions projects", however the specific privileges are not provides. For cancelling your personal project you need the view_personal_project permission + need to be the "Tellija" in that project (Based on the "Tellija" vaade wiki). But what permission do you need to cancel other types of projects (Project where the user is not the "Tellija")? Is it just the "VIEW_INSTITUTION_PROJECT_DETAIL" permission or do you also need some other permissions (Like RECEIVE_AND_MANAGE_PROJECT for example) ?

**Comment by MariusJulius** [comment]
@KaarelKa seems like it's missing there:
- VIEW_INSTITUTION_PROJECT_DETAIL (enables detail view, but not editing)
- For cancelling (managing): RECEIVE_AND_MANAGE_PROJECT  or MANAGE_PROJECT

**Comment by KaarelKa** [comment]
@MariusJulius Okay clear, but for user who is the "Tellija" of that project, only the "view_personal_project" privilege is needed right ?

**Comment by MariusJulius** [comment]
@KaarelKa yes, privileges "view_personal_projects" user is marked as "tellija" + order status limitations when it can be canceled.

**Comment by KaarelKa** [comment]
@PlaksoBirgit @MariusJulius just in case wanted to confirm that "personal projects" means that if the client, translation manager or vendor of a project is the logged in user, it is considered as a "personal project", but in all other cases it is considered an "institution project" ?

**Comment by PlaksoBirgit** [comment]
personal project is a project where the user is marked as the client or PM. The user must be logged in under the Institution to see their personal projects of that particular Institution (If they have other personal projects under another Institution then these are not seen)

**Comment by PlaksoBirgit** [comment]
1. Yes, correct
2. yes, this is good point, that if they assign someone else as the client then after saving (if no errors) they should be navigated away since it is not their project anymore. -> So we should add this distinction that if they don't have any privileges you mentioned and if they are not client or PM they should be navigated to landing page and in other case user stays in the project view. I'll add it to wiki

**Comment by KaarelKa** [comment]
@PlaksoBirgit @MariusJulius 

Hey have some extra things to confirm about privileges. 

a. Do I understand correctly that user can only view the project, if they:

1. Have the "VIEW_INSTITUTION_PROJECT_DETAIL" privilege 

2. **OR** are the "Client" of the project and have "VIEW_PERSONAL_PROJECT" **OR** "CHANGE_CLIENT" **OR** "CREATE_PROJECT" privilege

3.  **OR** Are the "Translation manager" of the project and have "RECEIVE_AND_MANAGE_PROJECT" **OR** "MANAGE_PROJECT" **OR** "VIEW_INSTITUTION_PROJECT_LIST" **OR** "VIEW_INSTITUTION_PROJECT_DETAIL" privileges


--- It's a bit unclear under "6.2 Flow: View institution's projects as PM" whether the 2 points under the preconditions are "AND" or "OR", so whether they need one of the "MANAGE" privileges **AND** one of the "VIEW" privileges or just any privilege out of this list. 


b. Do I understand correctly that the "Client" of a project can be changed only:

1. By the current "Client" of the project, if they have "CHANGE_CLIENT" permission and are editing the project.
2. By the person creating the project, if they have the "CHANGE_CLIENT" permission
3. By anyone with the "CHANGE_CLIENT" privilege, who also has either the "RECEIVE_AND_MANAGE_PROJECT" **OR** "MANAGE_PROJECT" privilege.


c. Do I understand correctly that all the other editable fields of a project can be changed by **Anyone** with either the "MANAGE_PROJECT" **OR** "RECEIVE_AND_MANAGE_PROJECT" privilege, regardless of whether they are the translation manager of that project ?

**Comment by PlaksoBirgit** [comment]
a) Viewing the project:

Precondition to navigate to Tellimused, Alamtellimused view (the only place to access project lists) is:   view_personal_project
 -> if addition to that privilege user is marked as client or PM in some project- they will be able to navigate to these projects
 -> if user additionally has privilege view_instituion_project_list - they see list of institution projects but CAN't view details
  -> if user additionally has privilege view_instituion_uncalimed_project_detail - they will be able to navigate to institution new projects
 -> if user additionally has privilege view_institution_project_detail - they will be able to navigate to all the institution projects
 
 b) Changing the client:
 1- yes, 2-yes, 3 - yes (manage_project privilege is needed-> the receive_project privilege is now separately and we don't check that when checking if user can edit or not)
 
 c) Editing project (and sub-projects) fields 
 correct, if user has sufficient privileges to navigate to the project detail view (see point a) then for any editable fields user needs the manage_project privilege 
 (the receive_project is now separate privilege)

**Comment by MariusJulius** [comment]
@VaabelMaret 
Order editing - tags not sent with PUT request

**Comment by VaabelMaret** [comment]
The tags are fixed.

**Comment by plakitkelly** [comment]
1) According to [Wiki](https://github.com/keeleinstituut/tv-tolkevarav/wiki/8.-Tellija-vaade-(Tellimuste-loomine,-korraldamine-ja-t%C3%B5lkimine-)#54-flow-edit-project-details-as-client) user with change_client privilege should be able to change client. 
But actually user needs 
-edit_vendor_db (for seeing another clients)
![image](https://github.com/keeleinstituut/tv-tolkevarav/assets/130659408/146a67b3-e313-4e8d-95cb-6baa5afd02e4)
-create_project (for clicking on save button)
privileges as well. @MariusJulius is this expected result that client must have edit_vendor_db privilege?

2) @VaabelMaret commented that tags are fixed but tags are not saved to "Tellimuse andmed" view and when editing the order, previously added tags are gone. (if user opens again the order or refreshes the page on order view, tags disappears)

**Comment by VaabelMaret** [comment]
@thenouan  @kadmit  
Currently, BE does not send selected tags back to a single project.

**Comment by MariusJulius** [comment]
> 1. According to [Wiki](https://github.com/keeleinstituut/tv-tolkevarav/wiki/8.-Tellija-vaade-(Tellimuste-loomine,-korraldamine-ja-t%C3%B5lkimine-)#54-flow-edit-project-details-as-client) user with change_client privilege should be able to change client.
>    But actually user needs
>    -edit_vendor_db (for seeing another clients)
>    ![image](https://user-images.githubusercontent.com/130659408/280771039-146a67b3-e313-4e8d-95cb-6baa5afd02e4.png)
>    -create_project (for clicking on save button)
>    privileges as well. @MariusJulius is this expected result that client must have edit_vendor_db privilege?

@plakitkelly user needs change_client privilege to change client, if user is PM (tõlkekorraldaja) he can change other fields. https://pundar.atlassian.net/wiki/spaces/TLKEV/pages/2149023753/Order+management+FE

**Comment by KaarelKa** [comment]
@MariusJulius Duplicate of https://github.com/keeleinstituut/tv-tolkevarav/issues/558

**Comment by plakitkelly** [comment]
OK
